broker/lib/cdn.py

```diff
@@ -12,3 +27,188 @@ def is_cdn_instance(service_instance) -> bool:
 
 def is_cdn_dedicated_waf_instance(service_instance) -> bool:
     return isinstance(service_instance, CDNDedicatedWAFServiceInstance)
+
+
+def parse_alarm_notification_email(instance, params):
+    if not is_cdn_dedicated_waf_instance(instance):
+        return None
+
+    return params.get("alarm_notification_email")
+
+
+def parse_cache_policy(params, cache_policy_manager: CachePolicyManager) -> str:
+    cache_policy = params.get("cache_policy", None)
+    if not cache_policy:
+        return None
+    if cache_policy not in config.ALLOWED_AWS_MANAGED_CACHE_POLICIES:
+        raise errors.ErrBadRequest(
+            f"'{cache_policy}' is not an allowed value for cache_policy."
+        )
+    managed_cache_policies = cache_policy_manager.get_managed_cache_policies()
+    if cache_policy in managed_cache_policies.keys():
+        return managed_cache_policies[cache_policy]
```

Comment:
I think that de-duplicating the logic from CachePolicyManager, this can be simplified to: 
```suggestion
def parse_cache_policy(params, cache_policy_manager: CachePolicyManager) -> str:
    cache_policy = params.get("cache_policy", None)
    return cache_policy_manager.get(cache_policy, None)
```

Reply:
6960f98d2286aaa9d3c31040a71bdb8775918d48